packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -602,26 +605,39 @@ class FloatingActionButton extends StatelessWidget {
         break;
     }
 
-    Widget result = RawMaterialButton(
-      onPressed: onPressed,
-      mouseCursor: _EffectiveMouseCursor(mouseCursor, floatingActionButtonTheme.mouseCursor),
+    final MaterialStateProperty<double> elevationValue = _EffectiveElevation(
       elevation: elevation,
       focusElevation: focusElevation,
       hoverElevation: hoverElevation,
-      highlightElevation: highlightElevation,
       disabledElevation: disabledElevation,
-      constraints: sizeConstraints,
-      materialTapTargetSize: materialTapTargetSize,
-      fillColor: backgroundColor,
-      focusColor: focusColor,
-      hoverColor: hoverColor,
-      splashColor: splashColor,
-      textStyle: extendedTextStyle,
-      shape: shape,
+      highlightElevation: highlightElevation,
+    );
+
+    Widget result = ElevatedButton(
+      style: ButtonStyle(
+        textStyle: MaterialStateProperty.all<TextStyle?>(extendedTextStyle),
+        backgroundColor: MaterialStateProperty.all<Color?>(backgroundColor),
+        foregroundColor: MaterialStateProperty.all<Color?>(foregroundColor),
+        overlayColor: _EffectiveOverlay(
+          hoverColor: hoverColor,
+          focusColor: focusColor,
+          splashColor: splashColor,
+        ),
+        elevation: elevationValue,
+        padding: MaterialStateProperty.all<EdgeInsetsGeometry?>(EdgeInsets.zero),
+        minimumSize: MaterialStateProperty.all<Size?>(Size(sizeConstraints.minWidth, sizeConstraints.minHeight)),
+        maximumSize: MaterialStateProperty.all<Size?>(Size(sizeConstraints.maxWidth, sizeConstraints.maxHeight)),
+        shape: MaterialStateProperty.all<OutlinedBorder?>(shape as OutlinedBorder),
```

Comment:
```suggestion
        shape: MaterialStateProperty.all<OutlinedBorder?>(shape as OutlinedBorder),
```

This is causing issues for customers that have 

```
FloatingActionButton(
  shape: Border(),
  ...
)
```

Is there a way that this can still be supported? Or is it a breaking change?

Reply:
It can still be supported by wrapping `ElevatedButton` with `DecoratedBox` BUT this can be done by the customer too, `floatingActionButton:` takes any widget so FAB can be wrapped in a `DecoratedBox`.

The reason when  `Border()`  provide to the `shape` parameter currently works is because `material.dart` checks if "shape" is a `ShapeBorder` otherwise based on material type, directly draw the border with custom paint hence there is no issue with this in the current design.


https://github.com/flutter/flutter/blob/75ee440329f555cac3d636a24bdc5c16c8ef60ef/packages/flutter/lib/src/material/material.dart#L908-L916

but there is no existing test where the border is provided to shape in the test class and looks a little bit off to me when the border is provided to shape parameter. 

If you still want this, I can wrap  `ElevatedButton `with `DecoratedBox` along with a couple of tests to make sure this behavior works as expected.

Reply:
> If you still want this, I can wrap ElevatedButton with DecoratedBox along with a couple of tests to make sure this behavior works as expected.

There are a surprising number of cases with Border, if we can still support it, it would be definitely be easier. Do you want to try it and see what it would look like?

Reply:
I just added the code. 
`decoration: ShapeDecoration(shape: shape),` is responsible for rendering the border and we're defaulting to a rectangular shape to match current behavior when a border is provided.

```dart
          shape: MaterialStateProperty.all<OutlinedBorder?>(shape is! OutlinedBorder
            ? const RoundedRectangleBorder()
            : shape as OutlinedBorder?,
          ),
```

![Screenshot 2022-08-22 at 11 25 48](https://user-images.githubusercontent.com/48603081/185952579-5574a5ba-d5e6-4b2d-89ea-97ac53f78226.png)